specifyweb/workbench/upload/upload_table.py

```diff
@@ -970,10 +970,13 @@ def _do_upload(
         concrete_field_changes = BoundUpdateTable._field_changed(
             reference_record, attrs
         )
+
+        # Edge case: Scope change is allowed for Loan -> division.
+        # See: https://github.com/specify/specify7/pull/5417#issuecomment-2613245552
         if any(
             scoping_attr in concrete_field_changes
             for scoping_attr in self.scopingAttrs.keys()
-        ):
+        ) and not (self.name == "Loan" and "division_id" in concrete_field_changes):
```

Comment:
nitpick:

Can you wrap this in a separate function that takes tablename and concrete_field_changes? Otherwise, it's quite distracting to see this (most of the logic in this file is generic). In the future, it would be easier to modify that function + we know what hacks we did